payroll/views.py

```diff
@@ -198,35 +198,25 @@ def get_context_data(self, **kwargs):
         return context
 
     def aggregate_department_statistics(self):
-        query = '''
-            SELECT
-              employer.name,
-              AVG(salary.amount) AS average,
-              SUM(salary.amount) AS budget,
-              COUNT(*) AS headcount,
-              employer.slug AS slug
-            {from_clause}
-            WHERE employer.parent_id = {id}
-            GROUP BY employer.id, employer.name
-            ORDER BY SUM(salary.amount) DESC
-        '''.format(from_clause=self.from_clause,
-                   id=self.object.id)
-
-        total_expenditure = sum(self.object.employee_salaries)
-
-        with connection.cursor() as cursor:
-            cursor.execute(query)
+        entity_base_pay = Sum(Coalesce("positions__jobs__salaries__amount", 0))
+        entity_extra_pay = Sum(Coalesce("positions__jobs__salaries__extra_pay", 0))
+        median_total_pay = Percentile(
+            (
+                NullIf(
+                    Coalesce("positions__jobs__salaries__amount", 0) +
+                    Coalesce("positions__jobs__salaries__extra_pay", 0), 0
+                )
+            ), 0.5, output_field=FloatField()
+        )
 
-            department_salaries = []
-            for department, average, budget, headcount, slug in cursor:
-                department_salaries.append({
-                    'department': department,
-                    'amount': average,
-                    'total_budget': budget,
-                    'headcount': headcount,
-                    'slug': slug,
-                    'percentage': (float(budget / total_expenditure) * 100)  # Percentage of total expenditure
-                })
+        department_salaries = Employer.objects.filter(parent_id=self.object.id)\
+                                              .values("name", "slug")\
+                                              .annotate(headcount=Count("positions__jobs__salaries"),
```

Comment:
For consistency, want to define a variable for headcount with this `Count()` above, like you did for pay aggregates?